File: src/p001/p001.jl
#!/usr/bin/env julia

function listcompsolution()
    sum(n for n in 3:999 if n % 3 ==0 || n % 5 == 0)
end

function seriessolution()
    function summultiplesof(num)
        numberofmultiples = div(999, num)
        div(numberofmultiples * (numberofmultiples + 1) * num, 2)
    end
    
    summultiplesof(3) + summultiplesof(5) - summultiplesof(15)
end

println(listcompsolution())
println(seriessolution())

File: src/p001/p001.ml
let sum_multiples_of n =
  let number_of_multiples = 999 / n in
  number_of_multiples * (number_of_multiples + 1) * n / 2

let series_solution = sum_multiples_of 3 + sum_multiples_of 5 - sum_multiples_of 15

let () = print_int series_solution
